### PR TITLE
Fixed: ZODB.Connection.TransactionMetaData didn't custom data

### PR DESCRIPTION
**Pull request by jimfulton**
storage that some storages rely on.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -2,6 +2,9 @@
  Change History
 ================
 
+- Fixed: ``ZODB.Connection.TransactionMetaData`` didn't custom data
```
Comment (jamadden) [inline CHANGES.rst]: Missing some words?

File: src/ZODB/Connection.py [hunk 2]
```diff
@@ -1321,3 +1321,22 @@ def _extension(self):
...
+    def data(self, ob):
+        try:
+            data = self._data
+        except AttributeError:
+            raise KeyError(ob)
+
+        try:
+            return data[id(ob)]
+        except KeyError:
+            raise KeyError(ob)
```
Comment (jamadden) [inline src/ZODB/Connection.py]: Is this simpler and equivalent (fewer lines of code to cover)?
```python
try:
   return self._data[id(ob)]
except (AttributeError, KeyError):
   raise KeyError(ob)
```

Not that it's a big deal. If you prefer the more explicit version that's fine by me.

Comment (jimfulton) [inline src/ZODB/Connection.py]: Sure. This code was copied verbatim from transaction.

File: CHANGES.rst (hunk 1 above)
Comment (jimfulton) [inline CHANGES.rst]: yup

**Pull request merged**